kornia/feature/disk/structs.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+
+
+@dataclass
+class DISKFeatures:
+    keypoints: Tensor
+    descriptors: Tensor
+    keypoint_logp: Tensor
+
+    def __post_init__(self):
+        assert self.keypoints.device == self.descriptors.device
+        assert self.keypoints.device == self.keypoint_logp.device
+
+    @property
+    def n(self):
+        return self.keypoints.shape[0]
+
+    @property
+    def device(self):
+        return self.keypoints.device
+
+    @property
+    def x(self):
+        return self.keypoints[:, 0]
```

Comment:
```suggestion
        return self.keypoints[..., 0]
```
does it make sense like this?

Reply:
It wouldn't be a bug, but the keypoints struct always holds a 2-dimensional tensor (i.e. not batched) and writing `...` implies it could be more than 2d which is not true.